### ISSUE TITLE
"error":"run Select error: invalid operation []map[string]interface {}

### ISSUE THREAD

**Issue by lesliechung88** [emqx/kuiper#166]
在执行如下查询时

select e.readings[0]->id from edgex_event_devicevirtual_stream AS e;

报这个错误：
kuiper > [{"error":"run Select error: invalid operation []map[string]interface {}([map[device:Temperature-Humidity-Device id:2f2c29c5-5d41-4e54-81fb-dc09050db1e7 name:Temperature origin:1584869772237825792 value:3.000000e+01]]) [] *xsql.BracketEvalResult(&{0 0})"}]
[{"error":"run Select error: invalid operation []map[string]interface {}([map[device:Temperature-Humidity-Device id:8a55d988-7a48-4fb3-9511-0d30deb681c6 name:Humidity origin:1584869773239062528 value:6.000152e+01]]) [] *xsql.BracketEvalResult(&{0 0})"}]
[{"error":"run Select error: invalid operation []map[string]interface {}([map[device:Temperature-Humidity-Device id:d4711724-3dd2-4f90-937e-6bb74b79a109 name:Temperature origin:1584869774243981824 value:3.000000e+01]]) [] *xsql.BracketEvalResult(&{0 0})"}]

我的流定义是：
edgex_event_devicevirtual_stream(id string, device string, origin bigint, readings array(struct(id string, origin bigint, device string, name string, value string)))
    WITH ( datasource = "edgex/device-virtual", FORMAT = "json", KEY = "id");

下面查询正常的：
select e.* from edgex_event_devicevirtual_stream AS e;
结果如下：
[{"device":"Temperature-Humidity-Device","id":"20c29095-1026-41e7-bb58-ee5f5f861f72","origin":1584868638793188608,"readings":[{"device":"Temperature-Humidity-Device","id":"4a2e19ba-7bdc-43c7-9c1c-5c3f5cf516f8","name":"Humidity","origin":1584868638786348032,"value":"3.401297e+01"}]}]

**Comment by jinfahua** [comment]
@lesliechung88 Now it's fixed in 0.2.1 version. You can use ``AS`` to specify an alias name.
```
select readings[0]->id as rid from edgex_event_devicevirtual_stream;
```

Otherwise, an auto-generated field name ``rengine_field_xx``  as below will be created for you. It's strongly NOT recommend to do so.

``` shell
kuiper > select readings[0]->id from edgex_event_devicevirtual_stream;
Query was submit successfully.
kuiper > [{"rengine_field_0":"4a2e19ba-7bdc-43c7-9c1c-5c3f5cf516f8"}]
```

Also, you could try EdgeX source that released in [0.2.1](https://github.com/emqx/kuiper/releases/tag/0.2.1), it should be much easier for you to use.

**Comment by lesliechung88** [comment]
@jinfahua Thinks,!  0.2.1 is OK!
